### PR TITLE
whoops I build an animation framework

### PR DESCRIPTION
**Pull request by yustarandomname**
Due the exelent state handling in Svelte 5 it is rather trivial to implement multi step animations in the applets. This is called in Layman's terms an: "animation framework".

animations are controled by the same mechanism as the other sliders before. An example can be inspected in the `src/routes/applet/testing/animation_studio/+page.svelte`

https://github.com/user-attachments/assets/9da0acb2-0ece-4e1a-b919-3b576bd9e1fe

> [!TIP]
> See applet: https://deploy-preview-245--playful-otter-9e0500.netlify.app/applet/matrix_operations/nilpotent

https://github.com/user-attachments/assets/ec367e4c-2925-408d-aa0c-13d251af38c5

> [!TIP]
> See applet 1: https://deploy-preview-245--playful-otter-9e0500.netlify.app/applet/linear_systems/row_reduction_1
> See applet 2: https://deploy-preview-245--playful-otter-9e0500.netlify.app/applet/linear_systems/row_reduction_2

### CODE REVIEW

**Comment by yustarandomname** [comment]
Beryl made something similar, have a look at this
- [ ] https://prime-applets.ewi.tudelft.nl/apps/linear-algebra/linear-systems/

**Comment by BEvanGelderen** [comment]
> @BEvanGelderen made something similar, have a look at this to see what is the overlap and differences
> 
> * [ ]  https://prime-applets.ewi.tudelft.nl/apps/linear-algebra/linear-systems/

actually, it were mainly Vera and Fokko and the other members of PGG at the time that created this animation. Marijn made the applet.

**Comment by ygeubbels** [comment]
- [ ] project on x2 axis ipv huidige step 2 tekst (volg tekst in het boek)
- [ ] Fokko is akkoord met de blauwe deur. Er is wel gesteggel over de kleur van de deurpost. 
- [ ]

**Comment by ygeubbels** [comment]
- [ ] middelpunt (als we de 4 getallen a, b, c en d noemen  (als ax + by + cz = d):
d / (a2 + b2 + c2) dot (a,b,c)

**Comment by ygeubbels** [comment]
- [ ] row reduction 1 (issue nr 50): omdat er geen 2D plaatje is moet er een QR code? komen op het pdf.
